lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -221,66 +222,12 @@ spec = do
 
-    describe "Iso8601Range" $ do
+    describe "Textual encoding" $ do
 
         describe "Can perform roundtrip textual encoding & decoding" $ do
-            textRoundtrip $ Proxy @(Iso8601Range "test-header")
-
-        let err = "Error encountered while decoding ISO 8601 time range: "
-
-        it "Cannot decode ranges without a space separator chararcter" $
-            property $ \(range :: Iso8601Range "test-header") -> do
-                let result = fromText @(Iso8601Range "test-header")
-                        $ T.filter (/= ' ')
-                        $ toText range
-                result `shouldBe` Left (TextDecodingError $ err
-                    <> "Unable to find required space separator character.")
-
-        it "Cannot decode ranges without a hyphen separator chararcter" $
-            property $ \(range :: Iso8601Range "test-header") -> do
-                let result = fromText @(Iso8601Range "test-header")
-                        $ T.filter (/= '-')
-                        $ toText range
-                result `shouldBe` Left (TextDecodingError $ err
-                    <> "Unable to find required hyphen separator character.")
-
-        it "Cannot decode ranges without a valid prefix" $
-            property $ \(range :: Iso8601Range "test-header-1") -> do
-                let result = fromText @(Iso8601Range "test-header-2")
-                        $ toText range
-                result `shouldBe` Left (TextDecodingError $ err
-                    <> "Invalid prefix string found. Expecting: "
-                    <> "\"test-header-2\"")
-
-        let exampleValidTime = "20080808T080808Z"
-
-        let exampleInvalidTimes =
-                [ ""
-                , "?"
-                , "2008"
-                , "20080808"
-                , "20080808T"
-                , "20080808T08"
-                , "20080808T0808"
-                , "20080808T080808" ]
-
-        describe "Cannot decode ranges without a valid start time" $
-            forM_ exampleInvalidTimes $ \invalidTime ->
-                it (T.unpack invalidTime) $ property $ do
-                    let text = "test " <> invalidTime <> "-" <> exampleValidTime
-                    fromText @(Iso8601Range "test") text
-                        `shouldBe` Left (TextDecodingError $ err
-                            <> "Invalid start time string: "
-                            <> show invalidTime)
-
-        describe "Cannot decode ranges without a valid end time" $
-            forM_ exampleInvalidTimes $ \invalidTime ->
-                it (T.unpack invalidTime) $ property $ do
-                    let text = "test " <> exampleValidTime <> "-" <> invalidTime
-                    fromText @(Iso8601Range "test") text
-                        `shouldBe` Left (TextDecodingError $ err
-                            <> "Invalid end time string: "
-                            <> show invalidTime)
```

Comment:
I think we still want some negative decoding paths for time strings, don't we ?

Reply:
> I think we still want some negative decoding paths for time strings, don't we ?

Indeed we do!

The `Iso8601Time` type reuses the functionality provided here: https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Data/Time/Text.hs

We actually already have quite a few negative decoding examples here:
https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/test/unit/Data/Time/TextSpec.hs#L196-L271.

We also check that equivalent times with different local timezones are decoded equivalently:
https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/test/unit/Data/Time/TextSpec.hs#L57

Do you feel the coverage of this test suite is adequate? Are there any gaps you would like to fill?